strum_macros/src/macros/enum_map.rs

```diff
@@ -0,0 +1,217 @@
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, quote};
+use syn::{spanned::Spanned, Data, DeriveInput, Fields, Ident};
+
+use crate::helpers::{non_enum_error, HasStrumVariantProperties};
+
+pub fn enum_map_inner(ast: &DeriveInput) -> syn::Result<TokenStream> {
+    let name = &ast.ident;
+    let gen = &ast.generics;
+    let vis = &ast.vis;
+    let mut doc_comment = format!("A map over the variants of [{}]", name);
+
+    if gen.lifetimes().count() > 0 {
+        return Err(syn::Error::new(
+            Span::call_site(),
+            "This macro doesn't support enums with lifetimes.",
+        ));
+    }
+
+    let Data::Enum(data_enum) = &ast.data else {
+        return Err(non_enum_error())
+    };
+    let map_name = syn::parse_str::<Ident>(&format!("{}Map", name)).unwrap();
```

Comment:
Is there a reason that you didn't use `format_ident!` here instead? It would avoid an unwrap and (imo) be cleaner.

Reply:
No, I didn't realize that was a thing. I thought I took that from EnumIter though.